odoo/orm/models.py

```diff
@@ -6322,7 +6329,7 @@ def __eq__(self, other):
         except AttributeError:
             if other:
                 warnings.warn(f"unsupported operand type(s) for \"==\": '{self._name}()' == '{other!r}'", stacklevel=2)
-        return NotImplemented
+        return False
```

Comment:
The purpose of `NotImplemented` was to enable a third-party object to be equal to a recordset.  Do you really want ot change that behavior?

Reply:
Reverted to not have behaviour change. Although, I am not sure how an object could be equal if it does not have `_id` or `_ids`.